### DISCUSSION TITLE
catch panic err="mkdir /data: file exists"

### DISCUSSION THREAD

**Discussion by bbista100** [erigontech/erigon#4014] [General]
Please help! I run "systemctl start supervisor" and I get this in the logs:
-------------
[EROR] [04-28|00:08:53.529] catch panic                              err="mkdir /data: file exists" stack="[main.go:22 panic.go:1038 flags.go:749 flags.go:573 flags.go:841 flags.go:889 node.go:79 main.go:37 app.go:526 app.go:286 main.go:27 proc.go:255 asm_amd64.s:1581]"
[INFO] [04-28|00:08:54.641] Build info                               git_branch=stable git_tag=v2022.04.04 git_commit=b35e87c84bc218f7af35ccff024f932d09f1051c
------------------

**Comment by AskAlexSharov** [comment]
/data is the file? Not a directory?

**Comment by AskAlexSharov** [comment]
ls -la /data

**Reply by bbista100** [reply]
Thanks for getting back!
It's a directory. Here is the ls -la /data output:
-------
lr@CRPT-U01:/$ ls -la /data
total 12
drwx------  3    lr            lr        4096 Apr 28 09:35 .
drwxr-xr-x 21  root       root    4096 Apr 28 09:33 ..
drwxr-xr-x  3  erigon    erigon 4096 Apr 28 09:35 erigon
-----------

**Reply by AskAlexSharov** [reply]
"systemctl start supervisor" - then show your erigon.service file

**Reply by AskAlexSharov** [reply]
OS? FS?

**Reply by bbista100** [reply]
Hi again. Ubuntu 22.04 LTS with Ext4 filesystem. The erigon.service files don't get created at all, I'm looking in /etc/systemd/system.

**Reply by bbista100** [reply]
Here is my /etc/supervisor/conf.d/erigon.conf file
--------------
[program:erigon]
command=bash -c '/opt/erigon/bin/erigon --datadir="/data/erigon/datadir" --private.api.addr="0.0.0.0:9090"'
user=erigon
autostart=true
autorestart=true
stderr_logfile=/var/log/supervisor/erigon.err.log
stderr_logfile_maxbytes=1000000
stderr_logfile_backups=10
stdout_logfile=/var/log/supervisor/erigon.out.log
stdout_logfile_maxbytes=1000000
stdout_logfile_backups=10
stopwaitsecs=300

**Reply by bbista100** [reply]
And my /etc/supervisor/conf.d/rpcdaemon.conf  file:
-----------------------
[program:rpcdaemon]
command=bash -c '/opt/erigon/bin/rpcdaemon --datadir="/data/erigon/datadir" --private.api.addr="localhost:9090" --http.addr="0.0.0.0" --http.port=8545 --http.vhosts="*" --http.corsdomain="*" --http.api="eth,debug,net,trace,web3,erigon" --ws'
user=erigon
autostart=true
autorestart=true
stderr_logfile=/var/log/supervisor/rpcdaemon.err.log
stderr_logfile_maxbytes=1000000
stderr_logfile_backups=10
stdout_logfile=/var/log/supervisor/rpcdaemon.out.log
stdout_logfile_maxbytes=1000000
stdout_logfile_backups=10

**Comment by bbista100** [comment]
And I've set this data directory for Erigon to work with: /data/erigon/datadir

**Reply by AskAlexSharov** [reply]
can you try to run without supervisord:
`/opt/erigon/bin/erigon --datadir=/data/erigon/datadir`
and show here the output

**Reply by AskAlexSharov** [reply]
FYI: now RPCDaemon can work inside Erigon. All that flags `--http.addr="0.0.0.0" --http.port=8545 --http.vhosts="" --http.corsdomain="" --http.api="eth,debug,net,trace,web3,erigon" --ws` are now available in Erigon also.

**Comment by bbista100** [comment]
Thanks Alex!

Here it is:
lr@CRPT-U01:/$ sudo /opt/erigon/bin/erigon --datadir=/data/erigon/datadir
[sudo] password for lr: 
INFO[04-29|01:09:48.460] Build info                               git_branch=stable git_tag=v2022.04.04 git_commit=b35e87c84bc218f7af35ccff024f932d09f1051c
INFO[04-29|01:09:48.461] Starting Erigon on Ethereum mainnet... 
INFO[04-29|01:09:48.461] Maximum peer count                       ETH=100 total=100
INFO[04-29|01:09:48.461] Set global gas cap                       cap=50000000
INFO[04-29|01:09:48.490] Opening Database                         label=chaindata path=/data/erigon/datadir/chaindata
INFO[04-29|01:09:49.285] Re-Opening DB in exclusive mode to apply migrations 
INFO[04-29|01:09:49.286] database closed                          label=chaindata
INFO[04-29|01:09:49.289] Apply migration                          name=db_schema_version
INFO[04-29|01:09:49.385] Applied migration                        name=db_schema_version
INFO[04-29|01:09:49.468] Updated DB schema to                     version=3.0.0
INFO[04-29|01:09:49.469] database closed                          label=chaindata
INFO[04-29|01:09:49.776] Writing custom genesis block             hash=0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3
INFO[04-29|01:09:49.876] Initialised chain configuration          config="{ChainID: 1 Homestead: 1150000 DAO: 1920000 DAOSupport: true EIP150: 2463000 EIP155: 2675000 EIP158: 2675000 Byzantium: 4370000 Constantinople: 7280000 Petersburg: 7280000 Istanbul: 9069000, Muir Glacier: 9200000, Berlin: 12244000, London: 12965000, Arrow Glacier: 13773000, Engine: ethash}"
INFO[04-29|01:09:49.876] Disk storage enabled for ethash DAGs     dir=/data/erigon/datadir/ethash-dags count=2
INFO[04-29|01:09:49.877] Initialising Ethereum protocol           network=1
INFO[04-29|01:09:49.877] Effective                                prune="--prune="
INFO[04-29|01:09:50.123] Starting private RPC server              on=127.0.0.1:9090
INFO[04-29|01:09:50.123] new subscription to logs established 
INFO[04-29|01:09:50.124] [1/16 Headers] Waiting for headers...    from=0
INFO[04-29|01:09:50.326] Mapped network port                      proto=tcp extport=30303 intport=30303 interface=NAT-PMP(192.168.0.1)
INFO[04-29|01:09:50.335] Mapped network port                      proto=udp extport=30303 intport=30303 interface=NAT-PMP(192.168.0.1)
INFO[04-29|01:09:50.485] Started P2P networking                   version=66 self=enode://4dfb6e3837f7207ca31c219f7c86bac068f1956eaf844e2fb9144d2276b9868f0d3dfce3a55c70f1104298f464879ed3061530c68ba06b221b2035b82eb9f7ba@172.16.0.2:30303 name=erigon/v2022.04.2-beta-b35e87c8/linux-amd64/go1.17.7
INFO[04-29|01:10:20.125] [1/16 Headers] Wrote block headers       number=23040 blk/second=768.000 alloc="183.93 MiB" sys="363.66 MiB"

**Reply by bbista100** [reply]
INFO[04-29|01:10:50.124] [1/16 Headers] Wrote block headers       number=379392 blk/second=11878.400 alloc="315.28 MiB" sys="533.20 MiB"
INFO[04-29|01:11:20.175] [1/16 Headers] Wrote block headers       number=966144 blk/second=19558.400 alloc="219.67 MiB" sys="752.32 MiB"
INFO[04-29|01:11:49.927] [p2p] GoodPeers                          eth66=10
INFO[04-29|01:11:50.125] [1/16 Headers] Wrote block headers       number=1500672 blk/second=17817.600 alloc="506.64 MiB" sys="752.32 MiB"
INFO[04-29|01:12:20.124] [1/16 Headers] Inserting headers         progress=2209650
INFO[04-29|01:12:50.124] [1/16 Headers] Wrote block headers       number=3108864 blk/second=53606.400 alloc="616.67 MiB" sys="1.28 GiB"
INFO[04-29|01:13:20.125] [1/16 Headers] Wrote block headers       number=3763200 blk/second=21811.200 alloc="897.44 MiB" sys="1.28 GiB"
INFO[04-29|01:13:49.927] [p2p] GoodPeers                          eth66=20
INFO[04-29|01:13:50.125] [1/16 Headers] Wrote block headers       number=4443648 blk/second=22681.600 alloc="1.06 GiB" sys="1.57 GiB"
INFO[04-29|01:14:20.124] [1/16 Headers] Wrote block headers       number=5305344 blk/second=28723.200 alloc="1.21 GiB" sys="1.73 GiB"
INFO[04-29|01:14:50.124] [1/16 Headers] Wrote block headers       number=6177792 blk/second=29081.600 alloc="1.30 GiB" sys="2.11 GiB"

**Reply by bbista100** [reply]
So it works like this, right? 

Now how do I change /etc/supervisor/conf.d/erigon.conf and /etc/supervisor/conf.d/rpcdaemon.conf  so "systemctl start supervisor" works too?

**Reply by AskAlexSharov** [reply]
yes, it works. supervisord - i don't know. read logs and docs of that systems.

**Reply by bbista100** [reply]
Thanks very much!
